File: Assets/Elephant/Plugins/iOS/Models/Hyperlink/Hyperlink.m
#import "Hyperlink.h"

@implementation Hyperlink

// MARK: - Initializers

- (instancetype)initWithMask:(NSString *)mask text:(NSString *)text url:(NSString *)url {
    self = [super init];
    
    if (self) {
        [self setMask:mask];
        [self setUrlReference:[text createHtmlUrlReferenceWithUrl:url]];
    }
    
    return self;
}


// MARK: - Methods

+ (NSString *)configurePopUpHtmlContentWithContent:(NSString *)content hyperlinks:(NSArray *)hyperlinks {
    content = [content stringByReplacingOccurrencesOfString:@"\n" withString:@"<br>"];
    
    for(int i = 0; i < hyperlinks.count; i++) {
        Hyperlink* hyperlink = [hyperlinks objectAtIndex:i];
        
        content = [content stringByReplacingOccurrencesOfString:[hyperlink mask] withString:[hyperlink urlReference]];
    }
    
    return content;
}

@end

File: Assets/Elephant/Plugins/Android/com/rollic/elephantsdk/Hyperlink/Hyperlink.java
package com.rollic.elephantsdk.Hyperlink;

import com.rollic.elephantsdk.Utils.StringUtils;

public class Hyperlink {
    public String mask;
    public String urlReference;

    public Hyperlink(String mask, String text, String url) {
        this.mask = mask;
        this.urlReference = StringUtils.createHtmlUrlReference(text, url);
    }
}
